### PR TITLE
[Cloud] Cloudstrapper upgrades for 1.5

### LINKED ISSUE #6672: [Cloudstrapper] Ubuntu AGW AMI Playbooks

### LINKED ISSUE #6909: Cloudstrapper for AGW Cluster

### PR DESCRIPTION
**Pull request by arunuke**
Signed-off-by: Arun Thulasi arunt@fb.com

## Summary

Upgrading Cloudstrapper to deploy 1.5

## Test Plan
Deploy orc8r and AGWs as specified in the README

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: experimental/cloudstrapper/README.md [hunk 1]
```diff
@@ -2,140 +2,178 @@
...
 
 ## 0. Prerequisites
-   - Create private repo on github to host helm charts. 
-     This information would be used in the build.yaml and cluster.yaml files when building and deploying Magma respectively. 
+
+   - Understand key directories and naming conventions
+     - CODE_DIR: Directory hosting Magma and Cloudstrapper code that was cloned for this purpose, typically in the ~/code/magma/experimental/cloudstrapper folder
+       - git clone <magma repo master>
+     - VARS_DIR: Directory where all variables reside, typically in the CODE_DIR/playbooks/roles/vars folder
+     - WORK_DIR: Directory created by administrator as part Cloudstrapper deployment where all working copies reside, typically in the ~/magma-experimental folder
```
Comment (karthiksubraveti) [inline experimental/cloudstrapper/README.md]: What does "created by administrator" mean here ? I see references to customers and administrators, should use same role either customer or user here ?

File: experimental/cloudstrapper/README.md [hunk 2]
```diff
@@ -2,140 +2,178 @@
...
      - AWS Access and Secret keys
-     - Github username and PAT (Personal Access Token)
-     - Dockerhub username and password 
 
-   - Understand key directories
-     - CODE_DIR: Directory hosting Magma code, typically in the ~/code/magma/experimental/cloudstrapper folder
-     - VARS_DIR: Directory where all variables reside, typically in the CODE_DIR/playbooks/roles/vars folder
-     - WORK_DIR: Directory where all working copies reside, typically n the ~/magma-experimental
-       folder
+   - If you are using the Marketplace experience,
```
Comment (karthiksubraveti) [inline experimental/cloudstrapper/README.md]: Marketplace experience term is introduced without any context. Should we link any additional document here for users to understand the market place experience.

File: experimental/cloudstrapper/README.md [hunk 3]
```diff
@@ -2,140 +2,178 @@
...
-   - Create private repo on github to host helm charts. 
-     This information would be used in the build.yaml and cluster.yaml files when building and deploying Magma respectively. 
+
+   - Understand key directories and naming conventions
+     - CODE_DIR: Directory hosting Magma and Cloudstrapper code that was cloned for this purpose, typically in the ~/code/magma/experimental/cloudstrapper folder
+       - git clone <magma repo master>
+     - VARS_DIR: Directory where all variables reside, typically in the CODE_DIR/playbooks/roles/vars folder
+     - WORK_DIR: Directory created by administrator as part Cloudstrapper deployment where all working copies reside, typically in the ~/magma-experimental folder
 
    - Identify two security keys in the Build/Gateway regions to be used for the following
```
Comment (karthiksubraveti) [inline experimental/cloudstrapper/README.md:11]: Context on why this step is necessary would be useful

File: experimental/cloudstrapper/README.md [hunk 4]
```diff
@@ -2,140 +2,178 @@
...
-     - CODE_DIR: Directory hosting Magma code, typically in the ~/code/magma/experimental/cloudstrapper folder
-     - VARS_DIR: Directory where all variables reside, typically in the CODE_DIR/playbooks/roles/vars folder
-     - WORK_DIR: Directory where all working copies reside, typically n the ~/magma-experimental
-       folder
+   - If you are using the Marketplace experience,
+     - Locate the Cloudstrapper AMI and version-specific AGW AMI and their respective AMI ids from Marketplace and make changes to cluster.yaml
+     - The AGW AMI id would be set to awsAgwAmi
+     - If using the Cloudstrapper tooling to provision the instance, the Cloudstrapper AMI id would be set to awsCloudstrapperAmi
+
+  - If you are building Magma by yourself or using your own repositories,
```
Comment (karthiksubraveti) [inline experimental/cloudstrapper/README.md]: Adding a link here on building and uploading helm charts and docker images will be useful

File: experimental/cloudstrapper/README.md [hunk 5]
```diff
@@ -2,140 +2,178 @@
...
-    ``` 
-  - Result: Created stackMantleEssentials with common security group, S3 storage
+  Before running the playbook, validate the following variables in defaults.yaml and cluster.yaml in VARS_DIR
 
-### 1.1 For users who do not have access to a Cloudstrapper AMI: Optional CI/CD
-  The devops playbooks are used to initialize a default instance, configure it to act as a Cloudstrapper and generate an
-  AMI that can be used as Cloudstrapper AMI and either published in the Marketplace as a public or community AMI or 
-  retained locally.
+  - defaults.yaml
+    - secgroupDefault indicates name of the default SecurityGroup that would be created and used across instances
```
Comment (karthiksubraveti) [inline experimental/cloudstrapper/README.md]: Can you add context on what instances this default security group is used?

File: experimental/cloudstrapper/README.md [hunk 6]
```diff
@@ -2,140 +2,178 @@
...
+    - bucketDefault indicates the name of the default S3 bucket used to persist information
```
Comment (karthiksubraveti) [inline experimental/cloudstrapper/README.md:52]: What is held in this s3 bucket?

File: experimental/cloudstrapper/README.md [hunk 7]
```diff
@@ -2,140 +2,178 @@
...
+    - stackEssentialsDefault indicates the name of the CloudFormation stack that hosts all the artifacts
+    - keyBoot refers to the key used for the Cloudstrapper instance. Use the keyCreate tag or if it's an existing key, use the right name
```
Comment (karthiksubraveti) [inline experimental/cloudstrapper/README.md:54]: if keyCreate is used can this step be skipped ?

**Review by karthiksubraveti — COMMENTED** [review]
Most changes look good. Left a few comments.

**Review by karthiksubraveti — APPROVED** [review]
Most changes look good. Left a few comments.

File: experimental/cloudstrapper/README.md (hunk 1 above)
Comment (arunuke) [reply experimental/cloudstrapper/README.md]: Standardized on user.

File: experimental/cloudstrapper/README.md (hunk 2 above)
Comment (arunuke) [reply experimental/cloudstrapper/README.md]: Changed the reference to indicate "previously built community Magma artifacts"

File: experimental/cloudstrapper/README.md (hunk 3 above)
Comment (arunuke) [reply experimental/cloudstrapper/README.md:11]: Cloudstrapper actually clones the chosen version of Magma and bakes it into the image. Removing the reference to the git clone step above.

File: experimental/cloudstrapper/README.md (hunk 4 above)
Comment (arunuke) [reply experimental/cloudstrapper/README.md]: Building is covered in Section 3 and I have added a reference to it.

File: experimental/cloudstrapper/README.md (hunk 5 above)
Comment (arunuke) [reply experimental/cloudstrapper/README.md]: Includes all EC2 instances created - Cloudstrapper and AGWs. I have added this reference now.

File: experimental/cloudstrapper/README.md (hunk 6 above)
Comment (arunuke) [reply experimental/cloudstrapper/README.md:52]: The s3 bucket is created to score any shared artifact. The current version does not use this, but future versions might use this to store terraform state, keys or any other stateful information with the necessary security privileges.

File: experimental/cloudstrapper/README.md (hunk 7 above)
Comment (arunuke) [reply experimental/cloudstrapper/README.md:54]: The name of the keys generated via keyCreate is actually picked up from defaults.yaml. Once created, other playbooks use this information to instantiate running EC2 instances for Cloudstrapper and AGWs. This step of updating defaults.yaml to reflect the name of the keys cannot be skipped. Added text to reflect that if two keys already exist and user will be using those keys, updating defaults.yaml to reflect those names and skipping keyCreate is adequate.

**Pull request merged**